### PR TITLE
ENH: support python 3.6

### PR DESCRIPTION
**Pull request by llllllllll**
I have been in 3.6 land too long without `%dis` or `%ast`. I figured I might as well fix the rest of the stuff while I was here.

### CODE REVIEW

File: codetransformer/transformers/literals.py [hunk 1]
```diff
@@ -161,6 +162,34 @@ def _build_map(self, instr):
...
 
+    if sys.version_info >= (3, 6):
+        def _construct_const_map(self, values, keys):
+            mapping = self.astype()
+            for key, value in zip(keys, values):
+                mapping[key] = value
+            return mapping
+
+        @pattern(instructions.LOAD_CONST, instructions.BUILD_CONST_KEY_MAP)
+        def _build_const_map(self, keys, instr):
```
Comment (llllllllll) [inline codetransformer/transformers/literals.py:173]: this is hit when a dictionary has all constant keys

File: codetransformer/transformers/precomputed_slices.py [hunk 2]
```diff
@@ -13,14 +13,14 @@ class precomputed_slices(CodeTransformer):
...
-    >>> dis(first_five)
+    >>> dis(first_five)  # doctest: +SKIP
       2           0 LOAD_FAST                0 (l)
                   3 LOAD_CONST               0 (None)
                   6 LOAD_CONST               1 (5)
                   9 BUILD_SLICE              2
                  12 BINARY_SUBSCR
                  13 RETURN_VALUE
-    >>> dis(precomputed_slices()(first_five))
+    >>> dis(precomputed_slices()(first_five))  # doctest: +SKIP
```
Comment (llllllllll) [inline codetransformer/transformers/precomputed_slices.py:23]: the instruction offsets changed, we may want to use the 3.6 versions

**Review by ssanderson — COMMENTED** [review]
LGTM

**Comment by llllllllll** [comment]
I will update the version and publish a pypi package soon

**Pull request merged**